### PR TITLE
Fix #2220 - Corrected some handling of relational inheritance queries.

### LINKED ISSUE #2220: Where() condition on properties in derived type after OfType() throws
Example tests expected to work in `InheritanceTestBase.cs` are:

        [Fact]
        public virtual void Can_use_of_type_kiwi_where_south_on_derived_property()
        {
            using (var context = CreateContext())
            {
                var animals = context.Set<Animal>().OfType<Kiwi>().Where(x => x.FoundOn == Island.South).ToList();

                Assert.Equal(1, animals.Count);
                Assert.IsType<Kiwi>(animals[0]);
                Assert.Equal(1, context.ChangeTracker.Entries().Count());
            }
        }

        [Fact]
        public virtual void Can_use_of_type_kiwi_where_north_on_derived_property()
        {
            using (var context = CreateContext())
            {
                var animals = context.Set<Animal>().OfType<Kiwi>().Where(x => x.FoundOn == Island.North).ToList();

                Assert.Equal(0, animals.Count);
                Assert.Equal(0, context.ChangeTracker.Entries().Count());
            }
        }

But the query call throws:

    System.ArgumentException : Property 'Microsoft.Data.Entity.FunctionalTests.TestModels.Inheritance.Island FoundOn' is not defined for type 'Microsoft.Data.Entity.FunctionalTests.TestModels.Inheritance.Animal'

Possibly related to the closed issue #1705, cc @anpete.

### PR DESCRIPTION
**Pull request by anpete**
@maumar

### CODE REVIEW

File: test/EntityFramework.Core.FunctionalTests/InheritanceTestBase.cs [hunk 1]
```diff
@@ -39,6 +41,40 @@ public virtual void Can_use_of_type_bird()
...
+                        .ToList();
+
+                Assert.Equal(1, animals.Count);
+                Assert.IsType<Kiwi>(animals[0]);
+                Assert.Equal(1, context.ChangeTracker.Entries().Count());
+            }
+        }
+
+        [Fact]
+        public virtual void Can_use_of_type_bird_with_projection()
```
Comment (maumar) [inline test/EntityFramework.Core.FunctionalTests/InheritanceTestBase.cs]: missing sql verification

File: test/EntityFramework.Core.FunctionalTests/InheritanceTestBase.cs [hunk 2]
```diff
@@ -187,6 +223,39 @@ var eagle
         }
 
         [Fact]
+        public virtual void Can_use_of_type_kiwi_where_south_on_derived_property()
```
Comment (maumar) [inline test/EntityFramework.Core.FunctionalTests/InheritanceTestBase.cs]: missing sql verification

File: test/EntityFramework.Core.FunctionalTests/InheritanceTestBase.cs [hunk 3]
```diff
@@ -187,6 +223,39 @@ var eagle
...
+                        .ToList();
+
+                Assert.Equal(1, animals.Count);
+                Assert.IsType<Kiwi>(animals[0]);
+                Assert.Equal(1, context.ChangeTracker.Entries().Count());
+            }
+        }
+
+        [Fact]
+        public virtual void Can_use_of_type_kiwi_where_north_on_derived_property()
```
Comment (maumar) [inline test/EntityFramework.Core.FunctionalTests/InheritanceTestBase.cs]: missing sql verification

**Comment by maumar** [comment]
:shipit:

File: test/EntityFramework.Core.FunctionalTests/InheritanceTestBase.cs (hunk 1 above)
Comment (anpete) [inline test/EntityFramework.Core.FunctionalTests/InheritanceTestBase.cs]: This is the base

**Pull request merged**